code/game/machinery/doors/door.dm

```diff
@@ -162,24 +169,18 @@
 	return try_to_activate_door(null, user)
 
 /obj/machinery/door/attack_tk(mob/user)
-	if(requiresID() && !allowed(null))
+	// allowed(null) will always return false, unless the door is all-access.
+	// So unless we've cut the id-scan wire, TK won't go through at all - not even showing an animation.
+	// But if we *have* cut the wire, this eventually falls through to attack_hand(), which calls try_to_activate_door(),
+	// which will fail because the door won't work if the wire is cut! Catch-22.
+	// Basically, TK won't work unless the door is all-access.
+	if(!id_scan_hacked() && !allowed(null))
 		return
 	..()
 
+/// Handles door activation via clicks, through attackby().
 /obj/machinery/door/proc/try_to_activate_door(obj/item/I, mob/user)
-	add_fingerprint(user)
-	if(operating || (obj_flags & EMAGGED))
-		return
-	if(!requiresID())
-		user = null //so allowed(user) always succeeds
-	if(allowed(user))
-		if(density)
-			open()
-		else
-			close()
-		return TRUE
-	if(density)
-		do_animate("deny")
+	return activate_door_base(user, TRUE)
```

Comment:
Why is this wrapper needed at all?

Reply:
It's not _needed_, and I can revert if you want. The issue is that the two functions had the same logic, but were written differently (different ordering of ifs, etc.) It made it harder to read, and much more likely that someone changing one might change both in a way that added subtle differences between the two.